### PR TITLE
MOTECH-2713: Error during creating new entity

### PR DESCRIPTION
**Pull request by jablonskidawid**
AllEntityDrafts.java: added superClass setting to draft - null value of this property caused error

V1__Base_version.sql: changed type of cannotBeInSet validation to string - there were some errors in UI

EntityServiceImpl.java: in getSchema method getting only actual entities, not drafts - drafts were saved without clicking "save changes" button

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: platform/mds/mds/src/main/resources/db/migration/mysql/V1__Base_version.sql [hunk 1]
```diff
@@ -457,7 +457,7 @@ INSERT INTO "TypeSettingOption" VALUES (1, 'REQUIRE'),
 INSERT INTO "TypeValidation" VALUES (1, 'mds.field.validation.minValue', 1),
                                     (2, 'mds.field.validation.maxValue', 1),
                                     (3, 'mds.field.validation.mustBeInSet', 2),
-                                    (4, 'mds.field.validation.cannotBeInSet', 1),
+                                    (4, 'mds.field.validation.cannotBeInSet', 2),
```
Comment (kmadej) [inline platform/mds/mds/src/main/resources/db/migration/mysql/V1__Base_version.sql]: please write new migration, do not change existing one. Moreover remember about postgres

File: platform/mds/mds/src/main/java/org/motechproject/mds/repository/internal/AllEntityDrafts.java [hunk 2]
```diff
@@ -68,7 +68,7 @@ public void setProperties(EntityDraft draft, Entity entity, String username) {
 
         draft.setDraftOwnerUsername(username);
         draft.setLastModificationDate(DateUtil.nowUTC());
-
+        draft.setSuperClass(entity.getSuperClass());
```
Comment (kmadej) [inline platform/mds/mds/src/main/java/org/motechproject/mds/repository/internal/AllEntityDrafts.java]: can you change in git history if this line existed in the past ? I'm wondering why it was not here, it always worth to check git history.

File: platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/EntityServiceImpl.java [hunk 3]
```diff
@@ -951,7 +951,7 @@ public SchemaHolder getSchema() {
         LOGGER.debug("Retrieving entities for processing");
 
         stopWatch.start();
-        List<Entity> entities = allEntities.retrieveAll();
+        List<Entity> entities = allEntities.getActualEntities();
```
Comment (kmadej) [inline platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/EntityServiceImpl.java]: can you check in git history if this method was used earlier here ? If yes, why someone changed it.

File: platform/mds/mds/src/main/java/org/motechproject/mds/repository/internal/AllEntityDrafts.java (hunk 2 above)
Comment (jablonskidawid) [inline platform/mds/mds/src/main/java/org/motechproject/mds/repository/internal/AllEntityDrafts.java]: It wasn't existed. I have checked history of this class.

File: platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/EntityServiceImpl.java (hunk 3 above)
Comment (jablonskidawid) [inline platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/EntityServiceImpl.java]: It wasn't used here. This function wasn't be changed after creation.

File: platform/mds/mds/src/main/java/org/motechproject/mds/repository/internal/AllEntityDrafts.java (hunk 2 above)
Comment (martokarski) [inline platform/mds/mds/src/main/java/org/motechproject/mds/repository/internal/AllEntityDrafts.java]: Do we really need to set this field?  I checked it and it worked without this line.

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
@kmadej Looks good to me

File: platform/mds/mds/src/main/resources/db/migration/default/V8__MOTECH-2713.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+UPDATE "TypeValidation" SET "TYPE_ID" = 2 WHERE "id" = 4;
```
Comment (kmadej) [inline platform/mds/mds/src/main/resources/db/migration/default/V8__MOTECH-2713.sql]: please add a brief comment to the migration file about what is happening here

**Comment by kmadej** [comment]
test this please

**Pull request merged**